packages/common/container-definitions/src/loader.ts

```diff
@@ -387,14 +388,17 @@ export interface IContainer extends IEventProvider<IContainerEvents> {
 	 * resulted in disposing it.
 	 */
 	dispose(error?: ICriticalContainerError): void;
+	dispose(disconnectReason?: DisconnectReason, error?: ICriticalContainerError): void;
 
 	/**
 	 * Closes the container.
 	 *
 	 * @param error - If the container is being closed due to error, this provides details about the error that
 	 * resulted in closing it.
+	 * @param disconnectReason - The reason for disconnecting the container
 	 */
 	close(error?: ICriticalContainerError): void;
+	close(disconnectReason?: DisconnectReason, error?: ICriticalContainerError): void;
```

Comment:
Will `error` always be defined if the disconnect reason is an error? Will it always be undefined otherwise? We might be able to make the typing a bit more explicit with more overloads.

Reply:
We can enforce this, I have updated the PR with error being required on Corruption